### ISSUE TITLE
[feature request] Add COCOB Optimizer

### ISSUE THREAD

**Issue by bhushan23** [pytorch/pytorch#9952]
COntinuous COin Betting (COCOB) in stochastic subgradient descent based algorithm that does not requires any learning rate.

Paper- https://arxiv.org/pdf/1705.07795.pdf

**Comment by bhushan23** [comment]
I will be taking up this task

**Comment by soumith** [comment]
we should add this to `contrib`, maybe it could be the first algo that goes into that package: https://github.com/pytorch/contrib

**Comment by HarshTrivedi** [comment]
I worked on analysing different coin betting optimization algorithms for parameter free optimization with Professor Francesco (@bremen) during his Convex Optimization [class](http://francesco.orabona.com/teaching.html#CSE592). I can't open full code repository since that work is unpublished yet, but [here](https://github.com/HarshTrivedi/coin-betting-optimizers-pytorch) is the minified repository that would be enough for this feature request. It contains cocob-backprop and code for 1 basic experiment for mnist and cifar. It also has small demo [here](https://github.com/HarshTrivedi/coin-betting-optimizers-pytorch) to see the betting game states in action on a given 1d function.

@bhushan23  feel free to open PR from that and let me know if anything is needed.

**Comment by bhushan23** [comment]
I had a chat with @HarshTrivedi 
given that he have already worked on COCOB and he wants to take this task, I am okay with it.
I will act as reviewer and work if something is blocking
@HarshTrivedi let me know if you need anything.
